awx/main/analytics/collectors.py

```diff
@@ -344,39 +380,56 @@ def _copy_table(table, query, path):
     return file.file_list()
 
 
-@register('events_table', '1.2', format='csv', description=_('Automation task records'), expensive=events_slicing)
+def jobevent_lastentry_cmp(lastentry):
```

Comment:
When `analytics/core.py` uses `max` to compare last entries for job events, we pass this key function in to ensure that pk-based entries come before partition table entries.